apps/src/templates/lessonOverview/LessonNavigationDropdown.jsx

```diff
@@ -0,0 +1,130 @@
+import React, {Component} from 'react';
+import i18n from '@cdo/locale';
+import Button from '@cdo/apps/templates/Button';
+import DropdownButton from '@cdo/apps/templates/DropdownButton';
+import color from '@cdo/apps/util/color';
+import {lessonShape} from '@cdo/apps/templates/lessonOverview/lessonPlanShapes';
+
+const styles = {
+  dropdown: {
+    display: 'inline-block'
+  },
+  boldText: {
+    fontFamily: '"Gotham 7r", sans-serif'
+  }
+};
+
+const LESSONS_PER_SECTION = 10;
+
+export default class LessonNavigationDropdown extends Component {
+  static propTypes = {
+    lesson: lessonShape.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    const currentLessonNumber =
+      props.lesson.unit.lessons.findIndex(l => l.key === props.lesson.key) + 1;
+    const sectionOfCurrentLesson = Math.ceil(
+      currentLessonNumber / LESSONS_PER_SECTION.toFixed(1)
+    );
+
+    this.state = {
+      currentSection: sectionOfCurrentLesson
+    };
+  }
+
+  linkWithQueryParams = link => {
+    const queryParams = window.location.search || '';
+    return link + queryParams;
+  };
+
+  handleDropdownClick = listItem => {
+    if (listItem.link) {
+      window.location.href = this.linkWithQueryParams(listItem.link);
+    } else {
+      this.setState({currentSection: listItem.sectionNumber});
+    }
+  };
+
+  createSectionsOfLessons = () => {
```

Comment:
nit: do you mind adding a comment about what the expected output of this function is? It took me a while to realize `sectionsAndLesson` would be a mix of expandable sections and linked lessons.